### ISSUE TITLE
More than 26 indices?

### ISSUE THREAD

**Issue by davidweichiang** [bdusell/semiring-einsum#31]
torch.einsum limits you to 26 indices but can semiring_einsum provide a non-string interface that allows more than 26?

**Comment by bdusell** [comment]
That would be a good idea. How about something like:

```python
compile_equation('abc,cde->bd')
```
can be written as
```python
compile_equation(inputs=[['a', 'b', 'c'], ['c', 'd', 'e']], output=['b', 'd'])
```
or
```python
compile_equation(inputs=[[0, 1, 2], [2, 3, 4]], output=[1, 3])
```

**Comment by davidweichiang** [comment]
That looks good to me. It's pretty close to numpy.einsum's alternative interface.

**Comment by ccshan** [comment]
Indeed, Python's common sequence operations make it easy to support
```
inputs=[['a', 'b', 'c'], ['c', 'd', 'e']], output=['b', 'd']
```
and
```
inputs=["abc", "cde"], output="bd"
```
with the same code.

**Comment by marawangamal** [comment]
Bumping this! Would be ultra helpful to incorporate the sublist notation to address this limitation

**Comment by bdusell** [comment]
Ok, I'll look into this when I have time.

**Comment by bdusell** [comment]
Added to the `master` branch!

**Comment by bdusell** [comment]
A couple of comments, FYI:
* It was already possible to use more than 26 variables with the string syntax by using characters besides a-z.
* PyTorch still has an upper limit of 64 tensor dimensions.
